code/game/objects/structures/flora.dm

```diff
@@ -109,21 +109,37 @@ GLOBAL_LIST_EMPTY(flora_uprooting_tools_typepaths)
 			user.visible_message(span_notice("[user] uproots [src]."),
 				span_notice("You uproot [src]."))
 			uproot(user)
-		W.play_tool_sound(src, 50)
+		used_item.play_tool_sound(src, 50)
 		return
 
-	if(!can_harvest(user, W))
+	if(!can_harvest(user, used_item))
 		return ..()
-	
+
+	//Checks specifically for FLORA_WOODEN to make sure the tool, if possible, is turned on. The can_harvest above makes sure this doesnt apply to all harvestables
+	if(used_item == /obj/item/melee/energy && used_item != /obj/item/melee/energy/axe) //energy axe is OP admin weapon and it interacts weirdly here
```

Comment:
Originally it DID check for sharpness, the PR that added that ugly item list removed that check, because there were lots of 'sharp' items that cant reasonably cut a tree.
Maybe itd be better with its own flag, like TREEFELLER or something? Regardless I'm scrapping the ugly list because checking item states from the harvest proc is disgusting